notification/app/notification/data/DataStore.scala

```diff
@@ -0,0 +1,30 @@
+package notification.data
+
+import aws.S3
+import models.TopicCount
+import play.api.Logger
+import play.api.libs.json.Format
+import utils.LruCache
+
+import scala.concurrent.duration.DurationLong
+import scala.concurrent.{ExecutionContext, Future}
+
+trait DataStore[T] {
+  def get()(implicit executionContext: ExecutionContext, format: Format[T]): Future[List[T]]
+}
+
+class S3DataStore[T](s3: S3[T]) extends DataStore[T] {
+  override def get()(implicit executionContext: ExecutionContext, format: Format[T]): Future[List[T]] = s3.fetch
+}
+
+class CachingDataStore[T](dataStore: DataStore[T]) extends DataStore[T] {
+  //There's only ever one list of topic counts
+  private val lruCache: LruCache[List[T]] = new LruCache[List[T]](1, 1, 5 minutes)
```

Comment:
I think given the nature of the state we hold, an AtomicValue might be better suited

Reply:
Yeah, good point. But do we want the limit the number off s3 hits we make?

Reply:
Actually you're right, we may just want to drop any cache on that.

Reply:
Yeah makes sense. So I'm going to drop this line here: https://github.com/guardian/mobile-n10n/blob/master/notification/app/notification/services/guardian/ReportTopicRegistrationCounter.scala#L15